### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by lhchavez**
Why
===

Tried another change, failed the CI.

What changed
============

This change makes the CI pass

Test plan
=========

```shell
yarn run lint
```

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { Channel } from './channel';
 import { EIOCompat } from './EIOCompat';
 import { ClientCloseReason, ChannelCloseReason } from './closeReasons';
 
+// eslint-disable-next-line no-shadow
```
Comment (lhchavez) [inline src/client.ts:10]: I have _no_ idea why eslint complains about this (and the other `no-shadow` one).

File: .eslintrc.js [hunk 2]
```diff
@@ -12,8 +12,15 @@ module.exports = {
...
     'import/prefer-default-export': 'off',
+    'import/extensions': [
+      'error',
+      {
+        ts: 'never',
+      },
+    ],
     indent: 'off',
     'max-len': ['error', { code: 120 }],
+    '@typescript-eslint/no-empty-function': 'off',
```
Comment (masad-frost) [inline .eslintrc.js:23]:
```suggestion
    '@typescript-eslint/no-empty-function': 'off',
    'no-shadow': 'off',
    '@typescript-eslint/no-shadow': 'error'
```

File: src/client.ts (hunk 1 above)
Comment (masad-frost) [inline src/client.ts:10]:
```suggestion
```
eslint rule is broken for typescript, I added a suggestion to the config. Let's remove this

File: package.json [hunk 3]
```diff
@@ -8,14 +8,14 @@
...
-    "lint": "eslint . --ext .js,.ts",
+    "lint": "eslint --ext .js,.ts src/ src.browser/",
     "build": "tsc",
     "build:browser": "yarn clean && yarn build && parcel build src.browser/index.js -d dist.browser --out-file crosis.js",
     "dev": "yarn build && (yarn build -- -w & parcel src.browser/index.html)",
     "prepublishOnly": "yarn clean && yarn build",
     "clean": "rm -rf dist dist.browser",
     "docs": "typedoc --out docs --name crosis --includeDeclarations --excludeExternals src/channel.ts src/client.ts src/closeReasons.ts node_modules/@replit/protocol/index.d.ts",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"no test specified. passing by vacuity\""
```
Comment (masad-frost) [inline package.json:18]: lol

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -12,15 +15,18 @@ jobs:
...
+
       - name: Setup Node.js environment
         uses: actions/setup-node@v2.1.2
 
       # Runs yarn install and caches the packages
       - uses: bahmutov/npm-install@v1
-      
+
+      - name: Build
+        run: yarn run build
```
Comment (masad-frost) [inline .github/workflows/main.yml:26]:
```suggestion
      - name: Types
        run: yarn run build --noEmit
```

**Review by masad-frost — APPROVED** [review]
Sweet thank you!

File: src/closeReasons.ts [hunk 5]
```diff
@@ -2,6 +2,7 @@ import { api } from '@replit/protocol';
 
 // In a separate file to circular dependencies in client and channel
 
+// eslint-disable-next-line no-shadow
```
Comment (masad-frost) [inline src/closeReasons.ts:5]:
```suggestion
```

File: .github/workflows/main.yml (hunk 4 above)
Comment (lhchavez) [reply .github/workflows/main.yml:26]: This needs to really build the files to be able to lint https://github.com/replit/crosis/blob/52333f3206bf08b716b5c043bc4be3cf2ce0cbdf/src.browser/index.js#L1

(not sure whether that's worth it, but i'll keep it as-is for the time being.

File: .eslintrc.js [hunk 6]
```diff
@@ -12,8 +12,17 @@ module.exports = {
...
+    'no-shadow': 'off',
+    '@typescript-eslint/no-shadow': 'error'
```
Comment (lhchavez) [inline .eslintrc.js:25]:
```suggestion
    '@typescript-eslint/no-shadow': 'error',
```

**Pull request merged**